File: java/parsers/src/test/java/jp/co/teruuu/parser/common/ManyParserTest.java
package jp.co.teruuu.parser.common;

import jp.co.teruuu.parser.common.ParseResult;
import jp.co.teruuu.parser.common.Parser;
import org.junit.Test;

import java.util.Arrays;
import java.util.List;

public class ManyParserTest {

    @Test
    public void test() {
        Parser<List<String>> manyParser = Parser.string("Hello").many();

        ParseResult result1 = manyParser.parse("HelloHelloHello");
        assert(result1 instanceof ParseResult.Success);
        assert(((ParseResult.Success) result1).value.equals(Arrays.asList("Hello", "Hello", "Hello")));

        ParseResult result2 = manyParser.parse("Hello HelloHello");
        assert(result2 instanceof ParseResult.Success);
        assert(((ParseResult.Success) result2).value.equals(Arrays.asList("Hello")));

        ParseResult result3 = manyParser.parse("helloHelloHello");
        assert(result3 instanceof ParseResult.Success);
        assert(((ParseResult.Success) result3).value.equals(Arrays.asList()));
    }
}

File: scala/parsers/src/test/scala/jp/co/teruuu/parser/common/ManyParserTest.scala
package jp.co.teruuu.parser.common

import org.scalatest.FlatSpec;

class ManyParserTest extends FlatSpec {

  "parse" should "successt" in {
    val parser = Parser.string("Hello").many()
    val result1 = parser.parse("HelloHello")
    assert(result1.isInstanceOf[ParseOk[_]])
    assert(result1.asInstanceOf[ParseOk[List[String]]].value.equals(List("Hello", "Hello")))

    val result2 = parser.parse("")
    assert(result2.isInstanceOf[ParseOk[_]])
    assert(result2.asInstanceOf[ParseOk[List[String]]].value.equals(List()))
  }

}
